draft-yorgos-dnsop-dry-run-dnssec.md

```diff
@@ -0,0 +1,302 @@
+%%%
+Title = "dry-run DNSSEC"
+abbrev = "dry-run-dnssec"
+docname = "@DOCNAME@"
+category = "std"
+area = "Internet"
+workgroup = "DNSOP Working Group"
+date = @TODAY@
+
+[seriesInfo]
+name = "Internet-Draft"
+value = "@DOCNAME@"
+stream = "IETF"
+status = "standard"
+
+[[author]]
+initials = "Y."
+surname = "Thessalonikefs"
+fullname = "Yorgos Thessalonikefs"
+organization = "NLnet Labs"
+[author.address]
+ email = "george@nlnetlabs.nl"
+[author.address.postal]
+ street = "Science Park 400"
+ city = "Amsterdam"
+ code = "1098 XH"
+ country = "Netherlands"
+
+[[author]]
+initials="W."
+surname="Toorop"
+fullname="Willem Toorop"
+organization = "NLnet Labs"
+[author.address]
+ email = "willem@nlnetlabs.nl"
+[author.address.postal]
+ street = "Science Park 400"
+ city = "Amsterdam"
+ code = "1098 XH"
+ country = "Netherlands"
+%%%
+
+
+.# Abstract
+
+This document describes a method called "dry-run DNSSEC" that allows for
+testing DNSSEC deployments without affecting the DNS service in case of
+configuration errors. It accomplishes that by introducing a new DS Type Digest
+Algorithm that signals to validating resolvers that dry-run DNSSEC is used for
+the zone. DNSSEC errors are then reported with DNS Error Reporting, but the
+bogus answer is withheld and instead resolvers fallback from dry-run DNSSEC and
+provide the answer they would normally produce for the zone. Further options
+are presented for testing key rollovers with this method and an option for
+clients to opt-in for dry-run DNSSEC errors and allow for end-to-end DNSSEC
+testing.
```

Comment:
We don't have a specific option for key-rollover testing, right? It is just a dry-run DS introducing the new algorithm I think.

```suggestion
bogus response is withheld. Instead resolvers fallback from dry-run DNSSEC and
provide the response that would have been answered without the presence of a dry-run DS.
A further option is presented for
clients to opt-in for dry-run DNSSEC errors and allow for end-to-end DNSSEC
testing.
```